uint8 solenoidState
---
uint32 retStat